src/main/java/sleepy/taskstorage/TaskList.java

```diff
@@ -0,0 +1,227 @@
+        switch (command) {
+        case "list":
+            printTasks();
+            break;
+        case "mark":
+            // Fallthrough
+        case "unmark":
+            // Fallthrough
+        case "delete":
+            int taskNumber = Integer.parseInt(parsedCommand[1]);
+            if (taskNumber < 1 || taskNumber > tasks.size()) {
+                throw new IllegalArgumentException("Invalid task number!");
+            }
+            handleTask(command, taskNumber);
+            break;
+        case "find":
+            String keywords = parsedCommand[1];
+            findTask(keywords);
+            break;
+        case "todo":
+            // Fallthrough
+        case "deadline":
+            // Fallthrough
+        case "plan":
+            // Fallthrough
+        case "event":
+            addTask(parsedCommand);
+            break;
+        default:
+            // Should never reach here - parser should have caught invalid command
+            assert false : "Parser did not parse command correctly";
+            throw new IllegalArgumentException("Incorrect command!");
+        }
+    }
+
+    /**
+     * Handles a task according to the operation type.
+     *
+     * @param operation Operation type.
+     * @param taskNumber Task number to be handled.
+     */
+    public void handleTask(String operation, int taskNumber) throws IllegalArgumentException {
+        assert taskNumber >= 1 && taskNumber <= tasks.size() : "Invalid task number!";
+        switch (operation) {
+        case "mark":
+            markTaskAsDone(taskNumber);
+            break;
+        case "unmark":
+            markTaskAsUndone(taskNumber);
+            break;
+        case "delete":
+            deleteTask(taskNumber);
+            break;
+        default:
+            // Should never reach here, parser should have detected invalid operation
+            assert false : "Parser did not parse command correctly";
+            throw new IllegalArgumentException("Invalid operation on a task!");
+        }
+    }
+
+    /**
+     * Searches for a task in this list by keyword(s), and responds with the match(es).
+     *
+     * @param keywords Keyword(s) used for the search.
+     */
+    public void findTask(String keywords) {
+        ResponseHandler.appendLineToResponse("Here are the matching tasks in your list:");
+        int i = 1;
+        for (Task task: tasks) {
+            if (task.getDescription().contains(keywords)) {
+                ResponseHandler.appendLineToResponse(i + "." + task.getDescription());
+                i++;
+            }
+        }
+    }
+
+    /**
+     * Creates and adds a task to this list.
+     *
+     * @param parsedTask Task to be added, in the form of a string.
+     * @throws IllegalArgumentException If the string does not contain a task type.
+     */
+    public void addTask(String[] parsedTask) throws IllegalArgumentException {
+        String taskType = parsedTask[0];
```

Comment:
A bit vague, as the variable is not a String but a collection of strings so should be plural

Reply:
Thank you for pointing this out, will fix it ASAP!